From 85a3d4955bbe3c7ef78dacbe14200083f50b635a Mon Sep 17 00:00:00 2001
From: Srinidhi Kasagar <srinidhi.kasagar@intel.com>
Date: Tue, 10 Mar 2015 13:47:30 +0530
Subject: [PATCH] mid: Do not apply d3_delay restriction to CHT platform

The d3_delay of 10ms recommended by pci spec need not be
applied even for CHT platforms like BYT platforms. Fix it.

Change-Id: Ifa4b66fdae1099d8581139c60c19c0292c4b2365
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-6903
Signed-off-by: Srinidhi Kasagar <srinidhi.kasagar@intel.com>
---
 arch/x86/platform/intel-mid/intel_mid_pcihelpers.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/x86/platform/intel-mid/intel_mid_pcihelpers.c b/arch/x86/platform/intel-mid/intel_mid_pcihelpers.c
index ba0e47f..1f97dff 100644
--- a/arch/x86/platform/intel-mid/intel_mid_pcihelpers.c
+++ b/arch/x86/platform/intel-mid/intel_mid_pcihelpers.c
@@ -191,7 +191,8 @@ static bool is_south_complex_device(struct pci_dev *dev)
  */
 static void pci_d3_delay_fixup(struct pci_dev *dev)
 {
-	if (platform_is(INTEL_ATOM_BYT)) {
+	if (platform_is(INTEL_ATOM_BYT) ||
+		platform_is(INTEL_ATOM_CHT)) {
 		/* All internal devices are in bus 0. */
 		if (dev->bus->number == 0 && is_south_complex_device(dev)) {
 			dev->d3_delay = INTERNAL_PCI_PM_D3_WAIT;
-- 
1.9.1

